### ISSUE TITLE
Overriding method does not match

### ISSUE THREAD

**Issue by HenriVi** [bmx-ng/bcc#308]
Hi,

this code generates an error as seen on title:

SuperStrict

Import brl.stream

Type a Extends TStream
        Method Close()
        EndMethod
EndType

As far as I see, the overriding method is exactly like the one declared in TIO.

-Henri

**Comment by woollybah** [comment]
The issue is that TStream is Strict, which defaults void return types to :Int, which the rest of the error message indicates.

Ideally, all official modules would be SuperStrict. Alternately, you could add :Int to your Close() definition. Or, make your code Strict instead of SuperStrict :-)

**Comment by woollybah** [comment]
I made the Stream modules SuperStrict...

**Comment by WaWannn** [comment]
Hi,

In SocketStream Module, 
Why TSocketStream extends TStream and the Close() overriding method works in Strict Mode, what's the difference ?

Cdlt,
Wawan

**Comment by HurryStarfish** [comment]
Both BRL.Stream and BRL.SocketStream were formerly Strict; now both of them are SuperStrict.
The difference is that in Strict mode, when you declare a variable, you can omit the type. If you do that, the type of the variable will be `Int`. The same holds true for the return types of functions.
In SuperStrict mode, you instead have to specify `:Int` (or `%`) just like any other variable type; not doing so is an error. You _can_ however omit the return type of a function in SuperStrict, which has a different meaning: the function can then return nothing at all, which in other languages is often called the "void" or "unit" return type.

tl;dr:
When you see `Method Close()` in Strict code, it means `Method Close:Int()`
When you see `Method Close()` in SuperStrict code, it means that the method returns nothing.
